chain/cosmos/chain_node.go

```diff
@@ -234,22 +233,25 @@ func (tn *ChainNode) WaitForBlocks(blocks int64) (int64, error) {
 
 		if deltaBlocks >= blocks {
 			tn.logger().
-				WithField("initialHeight", startingBlock).
-				Debugf("Time (sec) waiting for %d blocks: %d", blocks, i+1)
+				Debug("time waiting for blocks",
+					zap.Int64("numBlocks", blocks),
+					zap.Duration("duration", time.Duration(i+1)*time.Second),
+				)
 			return mostRecentBlock, nil // done waiting for consecutive signed blocks
 		}
 	}
 	return mostRecentBlock, errors.New("timed out waiting for blocks")
 }
 
 func (tn *ChainNode) maybeLogBlock(height int64) {
-	if tn.logger().Level() != "debug" {
+	entry := tn.logger().Check(zap.DebugLevel, "current block")
```

Comment:
This is a bit of an unusual pattern -- I see that we are trying to avoid the extra transactions involved if the logger is not enabled at debug level, and that makes sense. It throws me off to see the message `current block` declared once and then different fields depending on what exit path this function takes.

I think the more common pattern here would be:

```go
if !tn.logger().Core().Enabled(zap.DebugLevel) {
  return
}
// ...
if err != nil {
  tn.logger().Debug("Failed to get block", zap.Int64("height", height), zap.Error(err))
}
// ...
tn.logger().Debug("Current block", zap.Int64("height", height), zap.String("block", pp))
```

Reply:
Ah, thanks for that. I was looking for something similar, but the zap API did not make it obvious.